types/death/index.d.ts

```diff
@@ -51,16 +82,7 @@ declare function ON_DEATH(options: {
     SIGINT?: boolean;
     SIGTERM?: boolean;
     SIGQUIT?: boolean;
-    SIGHUP?: boolean;
-    uncaughtException?: boolean;
-}): (
-    callback: (
-        signal:
-            | "SIGINT"
-            | "SIGTERM"
-            | "SIGQUIT"
-            | "SIGHUP"
-            | "uncaughtException"
-    ) => void
-) => () => void;
+    uncaughtException: true;
+}): (callback: (signalOrErr: Signal | Error, origin?: string) => void) => () => void;
```

Comment:
Unfortunately, this is not guarnateed to be `Signal | Error`, since the `uncaughtException` handler reason can be any thrown value (despite even the Node.js docs saying the parameter type is `Error` 😱):

```js
process.on('uncaughtException', err => console.log(typeof err));
throw "asdfasdfasdf";
// > string

Reply:
should the type then be `any`?

Reply:
I prefer "Error", even though it's not 100% accurate, it's in line with the Node.js docs and is more accurate than the current type. But I'm also comfortable with "any" if you think that's best

Reply:
I think it’s up to you; this is why I didn’t explicitly request changes; I just wanted to make sure you’re aware and had thought about it.

- `unknown`: correct (all possible values in the type domain) but unhelpful (provides no hint about what the value is likely to be), cumbersome but very safe (must be narrowed to be used)
- `any`: correct but unhelpful, uncumbersome but very unsafe (can be used in any way without ceremony, even in ways that will not work at runtime)
- `Error`: incorrect but helpful (value will usually be an `Error`), uncumbersome but moderately unsafe (safety is better than `any` if the value is an `Error`, otherwise it’s a wash)

Reply:
The only reason I bring it up is that in TypeScript we don’t allow you to assume the value passed to a `catch` clause in a try/catch is an `Error` either. This feels analogous so I think is worth consideration. But I’ll go ahead and approve so you can decide what to do on your own.

Reply:
Makes sense. I'm gonna go ahead with 'Error'. It can typecast if the value ends up being anything else, and in most cases it will infact be an Error